### PR TITLE
Add APIs to send NMI and UART break

### PR DESCRIPTION
**Pull request by mkeeter**
This also adds UART break to the console, using the same key (Ctrl-A Ctrl-\) as `picocom`.

Needs to be tested on real hardware!

Closes https://github.com/oxidecomputer/hubris/issues/1068 and #51 once complete

TODO:
- [x] Plumb the GPIO NMI through `faux-mgs`

### CODE REVIEW

File: faux-mgs/src/usart.rs [hunk 1]
```diff
@@ -245,6 +263,13 @@ impl StdinOutBuf {
                         self.buf.push(c);
                     }
                 }
+                CTRL_BACKSLASH => {
+                    if self.in_prefix {
+                        return IngestResult::Break;
```
Comment (jgallagher) [inline faux-mgs/src/usart.rs:268]: This early return will cause any characters remaining in `buf` to be silently discarded. (The same is true on the early exit return, but we're exiting anyway.) Maybe we should have a `mut result` that we modify in this loop, and always run it to completion before returning?

File: gateway-sp-comms/src/single_sp.rs [hunk 2]
```diff
@@ -1070,6 +1070,20 @@ impl AttachedSerialConsoleSend {
...
+        let (tx, rx) = oneshot::channel();
+        self.inner_tx
+            .send(InnerCommand::Rpc(RpcRequest {
+                kind: MgsRequest::SerialConsoleBreak,
+                our_trailing_data: None,
+                response_tx: tx,
+            }))
+            .await
+            .unwrap();
+        rx.await.unwrap();
```
Comment (jgallagher) [inline gateway-sp-comms/src/single_sp.rs:1084]: Couple comments on this function:

1. There's a free `rpc` function that helps with some of the boilerplate (creating the oneshot, etc.).
2. We should check that the SP sent back the expected `SerialConsoleBreakAck`.

Combining both of those, I think this should be roughly:

```rust
        rpc(&self.inner_tx, MgsRequest::SerialConsoleBreak, None)
            .await
            .result
            .and_then(|(_peer, response, _data)| {
                response.expect_serial_console_break_ack()
            })
```

which will require adding `expect_serial_console_break_ack()` to `SpResponseExt`.

File: gateway-sp-comms/src/single_sp.rs [hunk 3]
```diff
@@ -1070,6 +1070,20 @@ impl AttachedSerialConsoleSend {
...
+        Ok(())
+    }
 }
```
Comment (jgallagher) [inline gateway-sp-comms/src/single_sp.rs:1087]: We should also add a `SingleSp::send_host_nmi()` method and a corresponding `faux-mgs` subcommand to call it. It doesn't necessarily have to be a part of this PR but might make it more convenient to test.

**Review by jgallagher — COMMENTED** [review]
Thanks for picking this up!

File: gateway-sp-comms/src/single_sp.rs (hunk 3 above)
Comment (mkeeter) [reply gateway-sp-comms/src/single_sp.rs:1088]: Yup, that was in my TODO list above; done in 89abdb393380383ef3c30e6a6e13757510eec73d

File: faux-mgs/src/usart.rs (hunk 1 above)
Comment (mkeeter) [reply faux-mgs/src/usart.rs]: That makes sense to me, done in 4aba91d7d4690abad4c1ada04610570f22cc1886

File: gateway-sp-comms/src/single_sp.rs (hunk 3 above)
Comment (jgallagher) [reply gateway-sp-comms/src/single_sp.rs:1088]: Ah sorry, missed the TODO list. My bad.

**Review by jgallagher — APPROVED** [review]

**Pull request merged**